### PR TITLE
chore(conversational-ai): internalize thread focus state and align stories

### PR DESCRIPTION
**Pull request by TarunAdobe**
## Description

- **`swc-conversation-thread`:** Removed public `active-index` / `activeIndex`; roving focus is owned by `FocusgroupNavigationController` only. Dropped the `focus()` fallback that focused a turn from cached index when `getActiveItem()` was null, and removed internal `_activeIndex` / `_syncActiveIndex` plumbing tied to that path.
- **`conversation-thread` stories:** Avoid styling `swc-conversation-thread` directly for layout (e.g. max width moved to a wrapper `div`); removed extra scroll-area inline padding used only for the full-pattern demo.
- **`swc-user-message`:** Removed host `margin-inline-end` gutter; trimmed the comment that described margin vs focus ring.

## Motivation and context

- Public `active-index` was not part of a clear consumer contract and duplicated controller state.
- Story-only padding / host inline styles suggested consumers had to style the thread to get correct layout or spacing.
- User-message end margin was easy to misread as required for correctness; spacing at the thread edge should come from intentional layout (container padding or design tokens), not hidden margin on the bubble host.

---

## Author's checklist

- [x] I have read the **[CONTRIBUTING](https://github.com/adobe/spectrum-web-components/blob/main/CONTRIBUTING.md)** and **[PULL_REQUESTS](https://github.com/adobe/spectrum-web-components/blob/main/PULL_REQUESTS.md)** documents.
- [x] I have reviewed at the Accessibility Practices for this feature, see: [Aria Practices](https://www.w3.org/TR/wai-aria-practices/)
- [x] I have added automated tests to cover my changes.
- [ ] I have included a well-written changeset if my change needs to be published.
- [ ] I have included updated documentation if my change required it.

---

## Reviewer's checklist

- [ ] Includes a Github Issue with appropriate flag or Jira ticket number without a link
- [ ] Includes thoughtfully written changeset if changes suggested include `patch`, `minor`, or `major` features
- [ ] Automated tests cover all use cases and follow best practices for writing
- [ ] Validated on all supported browsers
- [ ] All VRTs are approved before the author can update Golden Hash

### Manual review test cases

- [ ] **Conversation thread — keyboard**
  1. Storybook → **Conversational AI → Conversation thread** (Overview / Playground).
  2. Tab into the thread; use Arrow Up/Down, Home/End between turns.
  3. Expect one turn `tabindex="0"` at a time; focus moves predictably; no reliance on `active-index` in DOM.

- [ ] **Conversation thread — full pattern**
  1. Open **Full pattern** story.
  2. Send messages / scroll; tab between prompt and thread.
  3. Expect re-entry behavior unchanged; layout acceptable without story-only thread host padding.

- [ ] **User message — edge / focus**
  1. Stories with user bubbles flush to scroll edge (e.g. full pattern or narrow viewport).
  2. Focus a user turn / message.
  3. Expect focus ring not clipped; if it is, confirm product shell adds padding or we follow up with tokenized gutter in turn/thread CSS.

### Device review

- [ ] Did it pass in Desktop?
- [ ] Did it pass in (emulated) Mobile?
- [ ] Did it pass in (emulated) iPad?

## Accessibility testing checklist

- [ ] **Keyboard** (required — document steps below)
  1. Storybook → **Conversation thread** stories above.
  2. Tab from outside into thread; Arrow Up/Down between `swc-conversation-turn`; Tab into inner controls (e.g. feedback, sources) and back.
  3. Expect logical order, visible focus indicator on user/system content, no trap; behavior matches prior stories except no public `active-index`.

- [ ] **Screen reader** (required — document steps below)
  1. Same stories with VoiceOver / NVDA (or team default).
  2. Navigate by landmarks / groups through turns.
  3. Expect turn grouping labels (“User message” / “System message”) unchanged; no new duplicate announcements from thread host.

### CODE REVIEW

File: 2nd-gen/packages/swc/patterns/conversational-ai/conversation-thread/ConversationThread.ts [hunk 1]
```diff
@@ -62,52 +55,30 @@ export class ConversationThread extends SpectrumElement {
...
 
-  private _clampIndex(index: number, turns: HTMLElement[]): number {
-    if (!turns.length) {
-      return 0;
-    }
-
-    return Math.min(Math.max(index, 0), turns.length - 1);
-  }
-
   private _syncRovingFocusTarget(setLatestAsTabStop = false): void {
```
Comment (rubencarvalho) [inline 2nd-gen/packages/swc/patterns/conversational-ai/conversation-thread/ConversationThread.ts:78]: We're only calling this from an argument to this inside the `focus` and not passing an argument... so this `setLatestAsTabStop` will never be true

File: 2nd-gen/packages/swc/patterns/conversational-ai/conversation-thread/ConversationThread.ts [hunk 2]
```diff
@@ -62,52 +55,30 @@ export class ConversationThread extends SpectrumElement {
...
     this.focusgroupNavigationController.refresh();
     const turns = this._getItemsFromSlot();
     if (!turns.length) {
-      if (this.activeIndex !== 0) {
```
Comment (rubencarvalho) [inline 2nd-gen/packages/swc/patterns/conversational-ai/conversation-thread/ConversationThread.ts:108]: question: (i'm curious) what does the controller do if a focused/active turn is removed?
i know it's edge-casey, but could you add a test that removes the currently-active turn? I _think_ the tabindex may go to the last one (prob cause it's like focusout?).

Comment (TarunAdobe) [reply 2nd-gen/packages/swc/patterns/conversational-ai/conversation-thread/ConversationThread.ts:108]: How will this happen????

File: 2nd-gen/packages/swc/patterns/conversational-ai/conversation-thread/ConversationThread.ts (hunk 1 above)
Comment (TarunAdobe) [reply 2nd-gen/packages/swc/patterns/conversational-ai/conversation-thread/ConversationThread.ts:78]: syncRovingFocusTarget is also called from handleSlotChange, so setLatestAsTabStop can be true when new turns are appended while focus isn’t in the thread. The focus() path intentionally omits the argument so we only refresh and don’t force-jump to the last turn on every programmatic focus.

File: 2nd-gen/packages/swc/patterns/conversational-ai/conversation-thread/ConversationThread.ts (hunk 2 above)
Comment (rubencarvalho) [reply 2nd-gen/packages/swc/patterns/conversational-ai/conversation-thread/ConversationThread.ts:108]: idk, maybe you edit a sent message?

File: 2nd-gen/packages/swc/patterns/conversational-ai/conversation-thread/ConversationThread.ts (hunk 1 above)
Comment (rubencarvalho) [reply 2nd-gen/packages/swc/patterns/conversational-ai/conversation-thread/ConversationThread.ts:78]: aah didn't show up in the diff. okidoki!

**Review by rubencarvalho — APPROVED** [review]

**Review by Rajdeepc — APPROVED** [review]
LGTM!!!
